Source/Core/Core/PowerPC/PowerPC.h

```diff
@@ -57,6 +58,40 @@ struct TLBEntry
   u8 recent = 0;
 };
 
+struct PairedSingle
+{
+  u64 PS0AsIntegral() const { return ps0; }
+  u64 PS1AsIntegral() const { return ps1; }
+
+  double PS0AsDouble() const;
+  double PS1AsDouble() const;
+
+  void SetPS0(u64 value) { ps0 = value; }
+  void SetPS0(double value);
+
+  void SetPS1(u64 value) { ps1 = value; }
+  void SetPS1(double value);
+
+  void SetBoth(u64 lhs, u64 rhs)
+  {
+    SetPS0(lhs);
+    SetPS1(rhs);
+  }
+  void SetBoth(double lhs, double rhs)
+  {
+    SetPS0(lhs);
+    SetPS1(rhs);
+  }
+
+  void Fill(u64 value) { SetBoth(value, value); }
+  void Fill(double value) { SetBoth(value, value); }
```

Comment:
The primary motivation for keeping them separate names was to prevent accidentally using one set of functions over the other. The name Fill was chosen, because it, well, fills the paired single with a single value, just like how std::array's `fill` fills a whole sequence with a single value. I can change it to a different name if you have suggestions.

Reply:
Let's just leave it as-is I suppose...I can't really come up with a better name, and in hindsight `ps.SetBoth(valueForBoth)` (with just one parameter despite being called "both") doesn't sound too good either.